packages/lesswrong/lib/sql/UpdateQuery.ts

```diff
@@ -116,17 +161,47 @@ class UpdateQuery<T extends DbObject> extends Query<T> {
     format: (resolvedField: string, updateValue: Atom<T>[]) => Atom<T>[],
   ): Atom<T>[] {
     try {
+      const updateValue = this.compileUpdateExpression(value);
+
+      // If we're updating the value of a JSON blob without totally replacing
+      // it then we need to wrap the update in a call to `JSONB_SET`.
+      if (field.includes(".")) {
+        const {column, path} = this.buildJsonUpdatePath(field);
+        return format(
+          column,
+          ["JSONB_SET(", column, ",", path, ",", ...updateValue, ", TRUE)"],
+        );
+      }
+
       const resolvedField = this.resolveFieldName(field);
-      const updateValue = typeof value === "object" && value && Object.keys(value).some((key) => key[0] === "$")
-        ? this.compileExpression(value)
-        : [this.createArg(value)];
       return format(resolvedField, updateValue);
     } catch {
```

Comment:
Similar to before — this is catching all errors. Make sure that nothing you've done could conceivably cause an error for which the below warning would be erroneous.